csrc/host_ir/executor.h

```diff
@@ -13,30 +13,59 @@
 #include <host_ir/host_ir.h>
 #include <multidevice/communicator.h>
 #include <runtime/executor.h>
+#include <runtime/executor_abstract.h>
+#include <runtime/executor_params.h>
 #include <runtime/fusion_executor_cache.h>
 
 #include <c10/cuda/CUDAStream.h>
 
 namespace nvfuser {
 
+class HostIrExecutor : public ExecutorAbstract {
+ public:
+  HostIrExecutor(
+      int64_t fusion_id = 0,
+      int64_t concrete_id = 0,
+      int64_t runtime_id = 0,
+      int64_t group_id = 0);
+
+  static bool supported(Fusion* fusion);
+
+  void compile(Fusion* fusion);
+
+  bool isCompiled() const override;
+
+  NVF_API std::vector<at::Tensor> run(
+      KernelArgumentHolder& args,
+      std::vector<at::Tensor> outputs = {});
+
+  const std::unique_ptr<hir::HostIrContainer>& hostContainer() {
+    return host_ir_container_;
+  }
+
+ private:
+  std::unique_ptr<hir::HostIrContainer> host_ir_container_;
```

Comment:
nit:
```suggestion
  std::unique_ptr<hir::HostIrContainer> host_ir_container_ = nullptr;
```
since `isCompiled` relies on this

Reply:
I'll change it in isCompiled as for unique_ptr's we usually just condition on the object itself to know if it has been initialized.